tests/test_discovery.py

```
@@ -0,0 +1,265 @@
+
+    @pytest.mark.asyncio
+    async def test_discover_devices_skips_large_networks(self):
+        with patch.object(DeviceDiscovery, "scan_ip", return_value=None) as mock_scan:
+            discovery = DeviceDiscovery()
+            devices = await discovery.discover_devices(["10.0.0.0/8"])  # Too large
+            assert devices == []
+            mock_scan.assert_not_called()
+
+    @pytest.mark.asyncio
+    async def test_discover_devices_handles_scan_exception(self):
+        async def scan_raises(ip):
```

Ruff (0.15.0)

[warning] Unused function argument: `ip`

(ARG001)